### PR TITLE
[MERGE-WITH-SDK-ROLL-RC6]: BE changes to support propagating errors from SDK

### PR DESCRIPTION
**Pull request by chandrasekharan-zipstack**
## What

- Changes in test adapter exception to handle status code being passed from SDK
- Minor refactor in prompt-service while fetching context from vector DB to propagate errors with status code

## Why

- To correctly report status code of errors to avoid non 500 error logs and check relevant errors alone
- Error propagation for a few cases

## Can this PR break any existing features. If yes, please list possible items. If no, please explain why. (PS: Admins do not merge the PR without this section filled)

- No, minor error handling concerns addressed. However it should be merged only after the SDK release of `v0.54.0` / v0.54.0rc6` is done since it uses code from that

## Env Config

```
# Number of times to retry the /whisper-status API before failing the extraction
ADAPTER_LLMW_STATUS_RETRIES=5
```

## Related Issues or PRs

- https://github.com/Zipstack/unstract-sdk/pull/120

## Dependencies Versions

- `unstract-sdk` -> `v0.54.0` / v0.54.0rc6`

## Notes on Testing

- Tested with the SDK changes locally

## Screenshots

image

- Refer SDK PR for other screenshots

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

**Review by Deepak-Kesavan — APPROVED** [review]

File: backend/adapter_processor_v2/exceptions.py [hunk 1]
```diff
@@ -51,6 +52,20 @@ class TestAdapterError(APIException):
     status_code = 500
     default_detail = "Error while testing adapter"
 
+    def __init__(
+        self,
+        sdk_err: SdkError,
+        detail: Optional[str] = None,
+        code: Optional[str] = None,
+        adapter_name: Optional[str] = None,
```
Comment (gaya3-zipstack) [inline backend/adapter_processor_v2/exceptions.py:60]: I think , we could use kwargs for such details in future...

**Review by gaya3-zipstack — APPROVED** [review]
Looks good.

File: backend/adapter_processor_v2/exceptions.py (hunk 1 above)
Comment (chandrasekharan-zipstack) [reply backend/adapter_processor_v2/exceptions.py:60]: Will incorporate it next time I make similar changes, thanks for the suggestion

**Review by muhammad-ali-e — APPROVED** [review]
LGTM overall

**Pull request merged**